tests/functional_tests/mining.py

```diff
@@ -97,17 +105,55 @@ def mine(self, via_daemon):
         # wait till we mined a few of them
         target_height = prev_height + 5
         height = prev_height
-        timeout = 240 # randomx is slow to init
+        
+        """
+        Randomx init has high variance on CI machines due to noisy neighbors, 
+        taking up resources in parallel (including by our own jobs).
+        
+        Mining is organized in the following scheme:
+            1) first loop's pass: RandomX init and mining
+            2) every next pass:   only mining
+        Pass 1) takes much more time than pass 2)
+        Pass 1) uses all cores, pass 2) just one (currently)
+        For the above reasons both passes need separate timeouts and adjustments.
+        After the first pass, the timeout is being reset to a lower value.
+        """
+        
+        def get_timeout_int_float(time_pi, seconds_base, cores):
+            timeout_float = seconds_base * time_pi / cores
+            timeout_int = int(round(timeout_float + 1)) # +1 could be useful for smaller measurments
+            return timeout_int, timeout_float
+
+        timeout_base_init = 60 # RX init needs more time
+        timeout_base_mine = 20
+        cores_init = multiprocessing.cpu_count() # RX init uses all cores
+        cores_mine = res_status.threads_count    # Mining uses parametric number of cores
+        timeout_init, timeout_float_init = get_timeout_int_float(time_pi_all_cores,  timeout_base_init, cores_init)
+        timeout_mine, timeout_float_mine = get_timeout_int_float(time_pi_single_cpu, timeout_base_mine, cores_mine)
+        
+        msg = "Timeout for {} adjusted for the currently available CPU power, is"
```

Comment:
Why not use a format operation to inject both the operation name and the numeric value here? Using format to add a fixed value seems quite odd and unnecessary.

Reply:
Done.